src/targets/maven.ts

```diff
@@ -0,0 +1,163 @@
+import { TargetConfig } from '../schemas/project_config';
+import { releaseToMaven } from '../utils/mavenDeployment';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { BaseTarget } from './base';
+import { ConfigurationError } from '../utils/errors';
+import { withTempDir } from '../utils/files';
+import { isDryRun } from '../utils/helpers';
+import * as Github from '@octokit/rest';
+import {
+  getAuthUsername,
+  getGithubApiToken,
+  getGithubClient,
+  GithubRemote,
+} from '../utils/githubApi';
+import simpleGit from 'simple-git';
+import { homedir } from 'os';
+import { join } from 'path';
+import * as fs from 'fs';
+
+// TODO: add docs to the readme
+
+const GIT_REPO_OWNER = 'getsentry';
+const GIT_REPO_NAME = 'sentry-java';
+const FILES_TO_COMMIT = ['gradle.properties'];
+
+const USER_GRADLE_PROPS_FILE = join(homedir(), '/.gradle/gradle.properties');
```

Comment:
IMO, no. This file contains the secrets of Maven Central, so it shouldn't be in the project directory. If you see the [Gradle Build Environment](https://docs.gradle.org/current/userguide/build_environment.html), the choices if not using the project directory are 1) `$GRADLE_USER_HOME` and the default 2) `$USER_HOME/.gradle`; so I've added the option to use the former (instead of using directly the default value). This should allow more customization.